addons/google_account/google_account.py

```diff
@@ -89,14 +89,17 @@ def _get_google_token_json(self, cr, uid, authorize_code, service, context=None)
 
         headers = {"content-type": "application/x-www-form-urlencoded"}
 
-        try:
-            uri = self.get_uri_oauth(a='token')
-            data = werkzeug.url_encode(params)
+        if client_id and client_secret:
```

Comment:
@pedrobaeza thanks for the idea, that is indeed cleaner and stops the code even faster. To be absolutely sure.. you mean this right?
```
    def _refresh_google_token_json(self, cr, uid, refresh_token, service, context=None):  # exchange_AUTHORIZATION vs Token (service = calendar)
        res = False
        client_id = self.get_client_id(cr, uid, service, context)
        client_secret = self.get_client_secret(cr, uid, service, context)
        
        if not client_id or not client_secret:
            return res

        params = {
            'refresh_token': refresh_token,
            'client_id': client_id,
            'client_secret': client_secret,
            'grant_type': 'refresh_token',
        }

        headers = {"content-type": "application/x-www-form-urlencoded"}

        try:
            uri = self.get_uri_oauth(a='token')

            data = werkzeug.url_encode(params)
            st, res, ask_time = self._do_request(cr, uid, uri, params=data, headers=headers, type='POST', preuri='', context=context)
        except urllib2.HTTPError, e:
            if e.code == 400:  # invalid grant
                registry = openerp.modules.registry.RegistryManager.get(request.session.db)
                with registry.cursor() as cur:
                    self.pool['res.users'].write(cur, uid, [uid], {'google_%s_rtoken' % service: False}, context=context)
            error_key = json.loads(e.read()).get("error", "nc")
            _logger.exception("Bad google request : %s !" % error_key)
            error_msg = _("Something went wrong during your token generation. Maybe your Authorization Code is invalid or already expired [%s]") % error_key
            raise self.pool.get('res.config.settings').get_config_warning(cr, error_msg, context=context)
        return res
```

Ideally i'd like an else statement but that would mean a bigger diff too. 😄